### ISSUE TITLE
fontconfig@2.12.3 fails to build, known issue with gperf3.1

### ISSUE THREAD

**Issue by hartzell** [LLNL/spack#4546]
[edited w/ more info about working version]

Building fontconfig fails for me on CentOS 7.

```
[...]
  CC       fcname.lo                                                                                                                               CC       fcobjs.lo                                                                                                                             In file included from fcobjs.c:33:0:                                                                                                             fcobjshash.gperf:28:1: error: conflicting types for ‘FcObjectTypeHash’
 "aspect",FC_ASPECT_OBJECT
 ^                                                                                                                                               fcobjs.c:28:1: note: previous declaration of ‘FcObjectTypeHash’ was here                                                                          FcObjectTypeHash (register const char *str, register FC_GPERF_SIZE_T len);                                                                       ^                                                                                                                                               In file included from fcobjs.c:33:0:                                                                                                             fcobjshash.gperf:172:1: error: conflicting types for ‘FcObjectTypeLookup’                                                                        fcobjs.c:31:1: note: previous declaration of ‘FcObjectTypeLookup’ was here
 FcObjectTypeLookup (register const char *str, register FC_GPERF_SIZE_T len);
 ^
make[3]: *** [fcobjs.lo] Error 1                                                                                                                 make[3]: Leaving directory `/tmp/hartzelg/spack-stage/spack-stage-AKUZrk/fontconfig-2.12.3/src'                                                  make[2]: *** [all] Error 2                                                                                                                       make[2]: Leaving directory `/tmp/hartzelg/spack-stage/spack-stage-AKUZrk/fontconfig-2.12.3/src'
make[1]: *** [all-recursive] Error 1                                                                                                             make[1]: Leaving directory `/tmp/hartzelg/spack-stage/spack-stage-AKUZrk/fontconfig-2.12.3'                                                      make: *** [all] Error 2
```

fontconfig@2.12.1 works.

Google has a variety of hits for this problem, including a [patch here](https://chromium.googlesource.com/external/fontconfig/+/9878b306f6c673d3d6cd9db487f67eb426cc03df).

I'll dig more and see if I can come up with a solution.

**Comment by hartzell** [comment]
Additional info:

- https://bugs.freedesktop.org/show_bug.cgi?id=101280

   This thread does not contain a clean fix, just a hack-around.

- https://www.mail-archive.com/blfs-support@lists.linuxfromscratch.org/msg04104.html

  This thread let me to the patch referenced above.

**Comment by adamjstewart** [comment]
Closed by #4551.
